### PR TITLE
Quote escape bug fix

### PR DESCRIPTION
**Pull request by kahnvex**
Credit to @beck for telling me lodash templates were not the right choice. I've lifted code from gulp-static-i18n that needs to be modularized and made installable. This should fix most of the possible bugs we'd end up having.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,19 +14,31 @@ var extension = function(module, filename) {
...
-  var template = _.template(chunk.toString(), {
-    interpolate: opts.interpolate || /{{trans\s"([\s\S]+?)"}}/g
-  });
-
-  var translatedString = template(catalog)
+var escapeStr = function(str) {
+  return str
     .replace(/\n/g, '\\n')
     .replace(/"/g, '\\"');
+}
```
Comment (snitch) [inline index.js]: Missing semicolon.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: index.js [hunk 2]
```diff
@@ -14,19 +14,31 @@ var extension = function(module, filename) {
...
     .replace(/\n/g, '\\n')
     .replace(/"/g, '\\"');
+};
+
+var replaceText = function(catalog, opts, chunk, enc, callback) {
+  var template = chunk.toString();
+  var re = opts.interpolate || /\{\{trans\s*(?:"([^"]+)"|\'([^\']+)\')\s*\}\}/g;
+  var chunkString, match, msgid, needle, translated;
+
+  translated = template;
```
Comment (euwest) [inline index.js:28]: I don't see the _point_ of this(badum-tss)

Comment (euwest) [inline index.js:28]: actually i see the point, it's line 35 that doesn't make sense. i think it should be `match = re.exec(translated);`

Comment (euwest) [inline index.js:28]: nevermind, i just saw `Regex.prototype.exec()`'s successive match feature. that's dope

**Comment by beck** [comment]
I think this is better than using lodash (and it's probably hellafast) but now we're stepping into regex hell.  I think the only thing worse than regex'ing sourcecode is regex'ing for regexes.

Doing a quick browse of the handlebars source, I was hoping to find something we can pluck.  Then I saw their [quotedString()](https://github.com/wycats/handlebars.js/blob/d4070c36675bfecee290f20bd2d9c23a50e9e00b/lib/handlebars/compiler/code-gen.js#L101-L109) ... and I ran away.  Writing good parsers is hard.

I still recommend using handlebars and suffer the slower builds, but that is unfair of me to say since I'm not responsible for implementing this.  It's a call for @kahnjw but please consider the time of debugging regex vs slower builds.

**Comment by kahnvex** [comment]
> I still recommend using handlebars and suffer the slower builds, but that is unfair of me to say since I'm not responsible for implementing this. It's a call for @kahnjw but please consider the time of debugging regex vs slower builds.

It's not only that it is slower, but it is also that doing this would not allow consumers to use the Handlebars runtime in their application. We only want to replace translation tags, not every `{{ ... }}`.

**Comment by euwest** [comment]
`quotedString` looks like what we want for our escape function, actually...

**Comment by beck** [comment]
> We only want to replace translation tags, not every {{ ... }}.

Ahh, since this package is converting the template to a module, I assumed this package is responsible for the hbs > js compilation and any additional handlebar manipulation was nixed. My mistake.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: index.js [hunk 3]
```diff
@@ -41,8 +54,10 @@ var filterInterpolator = function(file, catalog, opts) {
 };
 
 var getCatalog = function(locale, localeDirs) {
-  var poParser = function(localeDir) {
-    var fp = path.join(localeDir, locale, 'LC_MESSAGES', 'messages.po');
+  var poParser = function(localeDir, index, localeDirs, en, filename) {
+    locale = en || locale;
+    var filename = filename || 'messages.po';
```
Comment (snitch) [inline index.js]: 'filename' is already defined.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by kahnvex** [comment]
Closing in favor of #6

**Pull request closed without merging**